Initium-Odp/src/com/universeprojects/miniup/server/commands/CommandSwitchCharacter.java

```diff
@@ -54,61 +54,47 @@ public void run(Map<String, String> parameters) throws UserErrorMessage {
 			direction = Integer.parseInt(parameters.get("direction"));
 			
 			//This method will return all characters that we can switch to. Filters out zambies and dead characters right off the bat.
-			List<CachedEntity> characters = db.getAlphabetSortedValidCharactersByUser(user.getKey());
 			
 			switch(direction) {
 			
-			//UP in characterswitcher.
 			case 1:
+			case 2:
+				List<CachedEntity> characters = db.getAlphabetSortedValidCharactersByUser(user.getKey());
+				
 				for(int i = 0; i < characters.size(); i++) {
-					CachedEntity c = characters.get(i);
-					//here, we are searching for the currentcharacter, so we know its index within the sorted list.
-					if(!GameUtils.equals(c.getKey(), currentCharacter.getKey)) continue;					
+					if(!GameUtils.equals(characters.get(0).getKey(), currentCharacter.getKey())) continue;
 					
-					//if we're at the first element in the list
-					if(i == 0) {
-						targetCharacterId = characters.get(characters.size()).getId();
+					//Move UP in character list.
+					if(direction == 1) {
+						//if we're at the front of the list, loop back to the back.
+						if(i == 0) {
+							targetCharacterId = characters.get(characters.size()).getId();
+							break;
+						}
```

Comment:
Will throw OOB exception, since you're trying to get the last array index + 1 (size() returns count, index is 0:size-1). 
Can simplify this processing step: `targetCharacterId = characters.get((i+ characters.size() - 1) % characters.size()).getId()`